src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -131,6 +134,8 @@ public void Reset()
             {
                 httpConnectionFeature.IsLocal = false;
             }
+
+            _currentITlsConnectionFeature = _tlsConnectionFeature;
```

Comment:
@cesarbs Is working on #365 which can make use of the same PrepareRequest method.